### PR TITLE
feature: add ability to stop a transaction

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request updates the `TransactionController` by adding a new method that stops a pending transaction by sending a 0 ETH transaction to the `from` address with a higher gas price.

### CODE REVIEW

**Comment by brunobar79** [comment]
@bitpshr Thanks for doing this. Before I look at it, Could you please clarify what's the situation between GABA and the TX controller in the extension?  I remember not so long ago you posted a standup update about saying you started to wrap the TX controller from the extension into GABA and that's also what was discussed during the last BE core meeting and AFAIK this feature it's already in the extension.

**Comment by bitpshr** [comment]
@brunobar79: The wrapping approach proved to be too difficult because of the way the extension controller relies on external events and external controller state. I wasn't able to satisfy both the external GABA API and the internal extension `TransactionController` requirements cohesively.

As it stands now, the GABA `TransactionController` is close to feature parity with the extension with about 15% of the loc. The only glaring omission at this point is transaction history. I think the right path forward is to bring GABA to full parity, and ultimately kill the extension version.

This PR gets us one step closer to that.

File: src/TransactionController.ts [hunk 1]
```diff
@@ -428,6 +430,37 @@ export class TransactionController extends BaseController<TransactionConfig, Tra
...
+		}
+
+		if (!this.sign) {
+			throw new Error('No sign method defined.');
+		}
+
+		const existingGasPrice = transactionMeta.transaction.gasPrice;
+		/* istanbul ignore next */
+		const existingGasPriceDecimal = parseInt(existingGasPrice === undefined ? '0x0' : existingGasPrice, 16);
+		const gasPrice = `0x${(existingGasPriceDecimal * 1.5).toString(16)}`;
```
Comment (bitpshr) [inline src/TransactionController.ts]: We multiply the previous gas price by `1.5`, but this is pretty arbitrary. I'm checking into what the extension does so we stay uniform...

Comment (brunobar79) [inline src/TransactionController.ts]: I'd move this to a constant at the top of the file with some clear name that explains what's doing.

**Pull request merged**